ethereum/p2p/src/main/java/tech/pegasys/pantheon/ethereum/p2p/network/DefaultP2PNetwork.java

```diff
@@ -741,4 +716,140 @@ private void onConnectionEstablished(final PeerConnection connection) {
     connections.registerConnection(connection);
     connectCallbacks.forEach(callback -> callback.accept(connection));
   }
+
+  public static class Builder {
```

Comment:
Not for this PR, but we should consider using a Builder library for these in the future to cut down on busy-work code.  Lombok and Guava autoValues both have support for the builder pattern like we use it.  Guava has the advantage we use part of it in our code already.